src/snowflake/cli/plugins/snowpark/models.py

```diff
@@ -65,6 +67,73 @@ def to_requirement_with_files(self):
         return RequirementWithFiles(self.requirement, self.files)
 
 
+@dataclass
+class RequirementWithWheelAndDeps:
+    """A dataclass to hold a requirement and corresponding .whl file."""
+
+    requirement: Requirement
+    wheel_path: Path | None
+    dependencies: List[str]
+
+    def extract_files(self, destination: Path) -> None:
+        if self.wheel_path is not None:
+            with zipfile.ZipFile(self.wheel_path, "r") as whl:
+                whl.extractall(destination)
+
+    def namelist(self) -> List[str]:
+        if self.wheel_path is None:
+            return []
+        with zipfile.ZipFile(self.wheel_path, "r") as whl:
+            return whl.namelist()
+
+
+@dataclass
+class WheelMetadata:
+    """A dataclass to hold metadata from .whl file.
+    [name] is the name of the package standarized accroding to
+    https://peps.python.org/pep-0491/#escaping-and-unicode
+    """
+
+    name: str
+    wheel_path: Path
+    dependencies: List[str]
+
+    @classmethod
+    def from_wheel(cls, wheel_path: Path):
+        """Parses wheel metadata according to
+        https://peps.python.org/pep-0491/#file-contents"""
+        with zipfile.ZipFile(wheel_path, "r") as whl:
+            metadata_path = [
+                path for path in whl.namelist() if path.endswith(".dist-info/METADATA")
+            ]
+            if len(metadata_path) != 1:
+                # malformatted wheel package
+                return None
+
+            root = zipfile.Path(whl)
+            metadata = (root / metadata_path[0]).read_text(encoding="utf-8")
+
+            dep_keyword = "Requires-Dist:"
+            dependencies = [
+                line[len(dep_keyword) :].strip()
+                for line in metadata.splitlines()
+                if line.startswith(dep_keyword)
+            ]
+            name = cls._get_name_from_wheel_filename(wheel_path.name)
+            return cls(name=name, wheel_path=wheel_path, dependencies=dependencies)
```

Comment:
Why not use one of the existing libraries?

Reply:
I wanted to avoid adding more external dependencies - this logic was simple enough to re-implement